MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/utils/CustomFirebaseMessagingService.java

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.mapboxandroiddemo.utils;
+
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+import com.mapbox.mapboxsdk.log.Logger;
+
+import java.util.Map;
+
+import static java.lang.Boolean.parseBoolean;
+
+/**
+ * This service is part of Firebase's Cloud Messaging setup to listen for incoming messages.
+ * With the right server side notification structure, the Maps SDK cache management methods
+ * can be hooked into the inbound notification data.
+ */
+public class CustomFirebaseMessagingService extends FirebaseMessagingService {
```

Comment:
Is this class practically used here?

Reply:
No, it's not. Will add a note that it's simply here as a solid starting point to finishing a serverside notification setup.